source/server/active_raw_udp_listener_config.h

```diff
@@ -0,0 +1,30 @@
+#pragma once
+
+#include "envoy/network/connection_handler.h"
+#include "envoy/registry/registry.h"
+#include "envoy/server/active_udp_listener_config.h"
+
+namespace Envoy {
+namespace Server {
+
+class ActiveRawUdpListenerFactory : public Network::ActiveUdpListenerFactory {
+  Network::ConnectionHandler::ActiveListenerPtr
+  createActiveUdpListener(Network::ConnectionHandler& parent,
+                          Network::ListenerConfig& config) const override;
+};
+
+// This class uses a protobuf config to create a UDP listener factory which
+// creates a Server::ConnectionHandlerImpl::ActiveUdpListener.
+// This is the default UDP listener if not specified in config.
```

Comment:
Do you think we should allow a default / unspecified, or should we deprecate that being allowed?  TODO for cleaning up in v3 fine by me.

Reply:
Allowing default/unspecified is for compatibility with current config. Ultimately, if the socket type is Datagram, this factory is better to be always explicitly specified in config.